### PR TITLE
Remove Database from Model

### PR DESCRIPTION
**Pull request by ryandialpad**
#### Type of PR:

- [ ] Bugfix
- [ ] Feature
- [x] Refactor
- [ ] Code style update
- [ ] Build-related changes
- [ ] Test
- [ ] Documentation
- [ ] Other, please describe:

#### Breaking changes:

- [ ] No
- [x] Yes

The following attributes have been removed from the `Model`:

- `_database`

The following methods have been removed from the `Model`:

- `$database`
- `$setDatabase`
- `$query`
- `$delete`
- `$deleteByKeyName`
- `$deleteByCompositeKeyName`

#### Purpose

The purpose of this PR is to remove `Database` access from the `Model` in order to simplify the `Model` and enforce a repository pattern when using the ORM. This means that we will stop relying on the `Model` to retrieve another `Model` and instead use the `Repository` or `Database` to fetch any other models.

#### Changes

- Removed `Database` from `Model` & any other associated methods
- Removed tests that tie back to the removed logic

### CODE REVIEW

File: test/feature/model/deletes_delete.spec.ts [hunk 1]
```diff
@@ -1,31 +0,0 @@
-import { createStore, fillState, assertState } from 'test/Helpers'
-import { Model, Attr, Str } from '@/index'
-
-describe('feature/model/deletes_delete', () => {
```
Comment (ryandialpad) [inline test/feature/model/deletes_delete.spec.ts:4]: The following method is deleted so we no longer need these tests.

File: test/feature/model/deletes_delete_composite_key.spec.ts [hunk 2]
```diff
@@ -1,34 +0,0 @@
-import { createStore, fillState, assertState } from 'test/Helpers'
-import { Model, Attr, Str } from '@/index'
-
-describe('feature/model/deletes_delete_composite_key', () => {
```
Comment (ryandialpad) [inline test/feature/model/deletes_delete_composite_key.spec.ts:4]: The following method is deleted so we no longer need these tests.

File: test/unit/model/Model.spec.ts [hunk 3]
```diff
@@ -8,10 +8,6 @@ describe('unit/model/Model', () => {
     @Attr() id!: number
   }
 
-  it('throws when accessing the database but it is not injected', () => {
-    expect(() => new User().$database()).toThrow()
-  })
```
Comment (ryandialpad) [inline test/unit/model/Model.spec.ts:13]: The database is no longer injected into the model so we no longer need this test.

File: test/unit/repository/Repository.spec.ts [hunk 4]
```diff
@@ -22,7 +22,6 @@ describe('unit/repository/Repository', () => {
     const user = store.$repo(User).make()
```
Comment (ryandialpad) [inline test/unit/repository/Repository.spec.ts:22]: Removing expectations of injected databases into the `Model`.

**Review by kiaking — APPROVED** [review]
AWESOOOME! Thanks a lot. Let's merge this after we make a new release for MorphOne and `withAll` and `withAllRecursive` methods. Since this is a breaking change, I think it's better to push plain new feature release first 👍

**Comment by ryandialpad** [comment]
> AWESOOOME! Thanks a lot. Let's merge this after we make a new release for MorphOne and `withAll` and `withAllRecursive` methods. Since this is a breaking change, I think it's better to push plain new feature release first 👍

Thanks!!!!! 🙌  🎉

**Review by kiaking — APPROVED** [review]

**Pull request merged**